src/lib.rs

```diff
@@ -1538,6 +1556,15 @@ impl PartialEq for SignatureAlgorithm {
 
 impl Eq for SignatureAlgorithm {}
 
+impl Hash for SignatureAlgorithm {
```

Comment:
Why not derive it?

Reply:
Clippy complains about deriving while `PartialEq` is implemented explicity: https://rust-lang.github.io/rust-clippy/master/index.html#derive_hash_xor_eq

Reply:
Understandable. Can you still add a comment on why there is no derive here?